File: backend/RetailSystem.Domain/Entities/Product.cs
namespace RetailSystem.Domain.Entities
{
    public class Product
    {
        public Guid Id { get; init; } = Guid.NewGuid();
        public required string ProductName { get; set; }
        public DateTimeOffset CreatedAt { get; init; } = DateTimeOffset.UtcNow;
        public decimal Price { get; set; }
        public decimal RatingAverage { get; set; } = 0;
        public ICollection<ProductVariant> ProductVariants { get; set; } = new List<ProductVariant>();
        public ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();
        public ICollection<Category> Categories { get; set; } = new List<Category>();
    }
}

File: web/src/features/product/types/product.ts
import type { UploadFile } from "antd";
import type { Category } from "../../category/types/category.types";

export interface Product {
  id: string;
  productName: string;
  description: string;
  price: number;
  createdAt: Date;
  updatedAt: Date;
  categories: Category[];
  productImages: ProductImage[];
  productVariants: ProductVariant[];
}

export interface ProductImage {
  id: string;
  imageUrl: string;
  isThumbnail: boolean;
}

export interface ProductVariant {
  id: string;
  productId: string;
  sizeId: string;
  stockQuantity: number;
}

export interface UploadProductSizeDto {
    id?: string;
    sizeId: string;
    quantity: number
}

export interface CreateProductRequest {
  productName: string;
  price: number;
  description: string | null;
  categoryIds: string[];
  productImages: UploadFile[];
  thumbnailIndex: number;
  sizesQuantity: UploadProductSizeDto[];
}

export interface UpdateProductRequest {
  id: string;
  productName: string;
  price: number;
  description: string | null;
  categoryIds: string[];
  removeImageIds: string[];
  productImages?: UploadFile[];
  thumbnailIndex?: number;
  thumbnailImageId?: string;
  sizesQuantity: UploadProductSizeDto[];
}